target_chains/aptos/contracts/sources/pyth.move

```diff
@@ -581,11 +713,280 @@ module pyth::pyth_test {
+            aptos_framework,
+            data_sources_for_test_vaa(),
+            100
+        );
+        pyth::update_price_feeds(vector[TEST_ACCUMULATOR_INVALID_ACC_MSG], coins);
+        cleanup_test(burn_capability, mint_capability);
+    }
+
+    #[test(aptos_framework = @aptos_framework)]
+    #[expected_failure(abort_code = 65564)]
+    fun test_accumulator_invalid_wormhole_message(aptos_framework: &signer) {
+        let (burn_capability, mint_capability, coins) = setup_accumulator_test(
+            aptos_framework,
+            data_sources_for_test_vaa(),
+            100
+        );
+
+        pyth::update_price_feeds(vector[TEST_ACCUMULATOR_INVALID_WH_MSG], coins);
+        cleanup_test(burn_capability, mint_capability);
+    }
+
+    #[test(aptos_framework = @aptos_framework)]
+    #[expected_failure(abort_code = 65539)]
+    fun test_accumulator_invalid_data_source(aptos_framework: &signer) {
+        let (burn_capability, mint_capability, coins) = setup_accumulator_test(aptos_framework, vector[data_source::new(
+            2, // correct emitter chain is 1
+            external_address::from_bytes(x"71f8dcb863d176e2c420ad6610cf687359612b6fb392e0642b0ca6b1f186aa3b")
+        )], 100);
+        pyth::update_price_feeds(vector[TEST_ACCUMULATOR], coins);
+        cleanup_test(burn_capability, mint_capability);
+    }
+
+    #[test(aptos_framework = @aptos_framework)]
+    fun test_accumulator_update_fee(aptos_framework: &signer) {
+        let (burn_capability, mint_capability, coins) = setup_accumulator_test(
+            aptos_framework,
+            data_sources_for_test_vaa(),
+            0
+        );
+        let single_update_fee = 50;
+        assert!(pyth::get_update_fee(&vector[
+            TEST_ACCUMULATOR,
+        ]) == single_update_fee, 1);
+
+        assert!(pyth::get_update_fee(&vector[
+            TEST_ACCUMULATOR,
+            TEST_ACCUMULATOR,
+        ]) == single_update_fee * 2, 1);
+
+        assert!(pyth::get_update_fee(&vector[
+            TEST_ACCUMULATOR,
+            TEST_ACCUMULATOR_3_MSGS,
+        ]) == single_update_fee * 4, 1);
+
+        assert!(pyth::get_update_fee(&vector[
+            TEST_ACCUMULATOR,
+            TEST_ACCUMULATOR_3_MSGS,
+            x"deaddeaddead", // random non-accumulator data
+        ]) == single_update_fee * 5, 1);
+
+        coin::destroy_zero(coins);
+        cleanup_test(burn_capability, mint_capability);
+    }
+
+    #[test(aptos_framework = @aptos_framework)]
+    #[expected_failure(abort_code = 65565)]
+    fun test_accumulator_invalid_proof(aptos_framework: &signer) {
+        let (burn_capability, mint_capability, coins) = setup_accumulator_test(
+            aptos_framework,
+            data_sources_for_test_vaa(),
+            100
+        );
+        pyth::update_price_feeds(vector[TEST_ACCUMULATOR_INVALID_PROOF_1], coins);
+        cleanup_test(burn_capability, mint_capability);
+    }
```

Comment:
I suggest adding a forward-compatibility check too where we generate a message payload that updates the minor version + extends all of the extensible parts of the message. (this can be done in a subsequent PR though)